### PR TITLE
Add helm resource-policy to CRDs to prevent deletion

### PR DESCRIPTION
**Pull request by Stono**
Add an annotation to all CRD's that will prevent helm from deleting them if tracked.

See https://github.com/istio/istio/issues/12451 for context

### CODE REVIEW

File: install/kubernetes/helm/istio-init/files/crd-11.yaml [hunk 1]
```diff
@@ -6,6 +6,8 @@ metadata:
     app: mixer
```
Comment (sdake) [inline install/kubernetes/helm/istio-init/files/crd-11.yaml]: I don't think these annotations are needed for crd11 files.

**Comment by Stono** [comment]
I've fixed the CLA issue

**Comment by duderino** [comment]
@costinm @sdake @ostromart can someone give this a lgtm (or ask for changes)

**Comment by morvencao** [comment]
Not sure whether this makes sense, because the CRDs are actually created by jobs, they should not be managed by the helm.

**Comment by sdake** [comment]
@morvencao I agree it is confusing and I'm not entirely sure why this works...  This is very non-intuitive.  However, it seems to work well enough.  See: https://github.com/istio/istio.io/issues/3679 for test results from 0.8->1.0.6->1.1.

Cheers
-steve

**Comment by sdake** [comment]
@Stono I had to make a new PR to address a few problems in your PR.  I did apply your original commit to the PR, but was for some reason unable to push to this branch.  My apologies for your work not being directly merged - we were keen to get an RC built.  Thanks again for the prototype and the suggestions - without you, we would have went down that rabbit hole of tiller-hack which would have been seriously painful.

Cheers
-steve

**Comment by Stono** [comment]
No problem at all @sdake ! Glad you got it sorted!

**Pull request closed without merging**